testlib/s3mock.py

```diff
@@ -72,77 +86,27 @@ def build_response(self, request):
         headers = self.resp["headers"]
         body = self.resp["body"]
 
-        response = Response()
-        response.status_code = status_code
-
-        if "content-type" not in headers:
-            headers["content-type"] = "text/xml"
-        if "content-length" not in headers:
-            headers["content-length"] = len(body)
-
-        response.raw = HTTPResponse(
+        response = HTTPResponse(
             body=io.BytesIO(body),
             headers=headers,
             status=status_code,
+            request_url=request.url,
+            request_method=request.method,
             reason=CODE_TO_REASON[status_code],
             preload_content=False,
             decode_content=False,
         )
-        response.reason = response.raw.reason
 
-        # From the request
-        response.url = request.url
-        response.request = request
+        if "content-type" not in headers:
+            headers["content-type"] = "text/xml"
+        if "content-length" not in headers:
+            headers["content-length"] = len(body)
 
-        response.connection = self
+        response.request = request
 
         return response
 
 
-class FakeAdapter(BaseAdapter):
```

Comment:
This is how we "mocked" things in requests. With urllib3, we're just patching urlopen instead, so we don't need this anymore. I folded the necessary bits into `S3Mock`.